campaign.go

```diff
@@ -0,0 +1,144 @@
+package stream_chat
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+type CampaignStatusName string
+
+const (
+	StatusDraft      CampaignStatusName = "draft"
+	StatusReady      CampaignStatusName = "ready"
+	StatusStopped    CampaignStatusName = "stopped"
+	StatusScheduled  CampaignStatusName = "scheduled"
+	StatusCompleted  CampaignStatusName = "completed"
+	StatusFailed     CampaignStatusName = "failed"
+	StatusInProgress CampaignStatusName = "in_progress"
+)
+
+type SegmentFilter struct {
+	Channel map[string]interface{} `json:"channel,omitempty"`
+	User    map[string]interface{} `json:"user,omitempty"`
+}
+
+type CampaignData struct {
+	SegmentID         string            `json:"segment_id,omitempty"`
+	SenderID          string            `json:"sender_id,omitempty"`
+	ChannelType       string            `json:"channel_type,omitempty"`
+	Text              string            `json:"text,omitempty"`
+	Defaults          map[string]string `json:"defaults,omitempty"`
+	Filter            SegmentFilter     `json:"filter,omitempty"`
+	Attachments       []*Attachment     `json:"attachments,omitempty"`
+	PushNotifications bool              `json:"push_notifications,omitempty"`
```

Comment:
PushNotifications will be removed, you can assume this and remove it

Reply:
actually, we are actively working on campaigns, i suspect changes may come in the future (not 100% sure but there is a risk)), i think better to postpone this PR a bit until we are 100% sure on API contract. @ferhatelmas what do you think?

Reply:
sg, good to handle untyped languages first

Reply:
👍 those are done.
closing the issue.